### ISSUE TITLE
NVDA version alpha-31360,dd44cbb7: In the "Regular expression for text paragraph navigation" field, NVDA does not announce all the characters

### ISSUE THREAD

**Issue by paulber19** [nvaccess/nvda#16288]
### Actual behavior:

- Open "NVDA settings",
- go to "Advanded" cateNngory and check "I understand that changing these settings may cause NVDA to function incorrectly. check box.
- pess "shift+tab" to reach "Regular expression for text paragraph navigation" field.
- press "end" and press "leftArrow" key some times.

After the third time,characters are not announced by NVDA.

[.…]{1,3}["”»)]?(\[\d+\])*(?=[\r\n  ]|$)|[?!]|[．！？：；])
### Expected behavior:

NVDA must announce the characters
#### NVDA installed/portable/running from source:
NVDA portable
#### NVDA version:
NVDA version alpha-31360,dd44cbb7: 
#### Windows version:
Windows 10
#### Name and version of other software in use when reproducing the issue:
None
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
No
#### If NVDA add-ons are disabled, is your problem still occurring?
Yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Yes

**Comment by Adriani90** [comment]
Is the Window maximized?
We also need a log file at debug logging level.

**Comment by CyrilleB79** [comment]
@paulber19, could you modify the title of this issue to provide it in English?
(not for me but for other people)

**Comment by wmhn1872265132** [comment]
This may be caused by some full-width symbols not being defined in the corresponding language's symbol file, and the speech engine not recognizing the symbol.

**Comment by paulber19** [comment]
Yes and to clarify, the characters are sent to the synthesizer (OneCore 
or Eloquence) (see the log below), but nothing is spoken.
There are 5 characters (or symbols) starting from position 70 in the string:
((?<=\w)(?<!\d)["”»)]?[.…]{1,3}["”»)]?(\[\d+\])*(?=[\r\n  ]|$)|[?!]| 
[．！？：；])

These are:
65294 (0xff0e)
65281 (0xff01)
65311 (0xff1f)
65306 (0xff1a)
65307 (0xff1b)

Here is the end of the log:
IO - inputCore.InputManager.executeGesture (08:35:33.616) - winInputHook 
(3396):
Input: kb(desktop):shift+tab
IO - speech.speech.speak (08:35:33.830) - MainThread (1140):
Speaking ['Avancé', 'page de propriété', "Avertissement !\nLes réglages 
suivants s'adressent aux utilisateurs avancés. Les modifier peut 
provoquer un fonctionnement incorrect de NVDA. Veuillez ne les changer 
que si vous savez ce que vous faites ou si les développeurs de NVDA vous 
l'ont spécifiquement demandé.", CancellableSpeech (still valid)]
IO - speech.speech.speak (08:35:33.845) - MainThread (1140):
Speaking ['Regular expression for text paragraph navigation', 'édition', 
CancellableSpeech (still valid), 'sélectionné 
((?<=\\w)(?<!\\d)["”»)]?[.…]{1,3}["”»)]?(\\[\\d+\\])*(?=[\\r\\n 
\xa0]|$)|[?!]|[．！？：；])']
IO - inputCore.InputManager.executeGesture (08:35:36.436) - winInputHook 
(3396):
Input: kb(desktop):end
DEBUG - editableText.EditableText._hasCaretMoved (08:35:36.481) - 
MainThread (1140):
Caret move detected using event. Elapsed 0 sec, retries 0
IO - speech.speech.speak (08:35:36.496) - MainThread (1140):
Speaking ['vide']
IO - speech.speech.speak (08:35:36.511) - MainThread (1140):
Speaking 
['((?<=\\w)(?<!\\d)["”»)]?[.…]{1,3}["”»)]?(\\[\\d+\\])*(?=[\\r\\n 
\xa0]|$)|[?!]|[．！？：；]) non sélectionné']
IO - inputCore.InputManager.executeGesture (08:35:38.276) - winInputHook 
(3396):
Input: kb(desktop):leftArrow
DEBUG - editableText.EditableText._hasCaretMoved (08:35:38.321) - 
MainThread (1140):
Caret move detected using event. Elapsed 0 sec, retries 0
IO - speech.speech.speak (08:35:38.336) - MainThread (1140):
Speaking ['parenthèse droite', EndUtteranceCommand()]
IO - inputCore.InputManager.executeGesture (08:35:38.881) - winInputHook 
(3396):
Input: kb(desktop):leftArrow
DEBUG - editableText.EditableText._hasCaretMoved (08:35:38.926) - 
MainThread (1140):
Caret move detected using event. Elapsed 0 sec, retries 0
IO - speech.speech.speak (08:35:38.937) - MainThread (1140):
Speaking ['crochet droit', EndUtteranceCommand()]
IO - inputCore.InputManager.executeGesture (08:35:39.426) - winInputHook 
(3396):
Input: kb(desktop):leftArrow
DEBUG - speech.manager.SpeechManager._handleIndex (08:35:39.426) - 
MainThread (1140):
Unknown index 28, speech probably cancelled from main thread.
DEBUG - editableText.EditableText._hasCaretMoved (08:35:39.466) - 
MainThread (1140):
Caret move detected using event. Elapsed 0 sec, retries 0
IO - speech.speech.speak (08:35:39.481) - MainThread (1140):
Speaking [CharacterModeCommand(True), '；', EndUtteranceCommand()]
IO - inputCore.InputManager.executeGesture (08:35:39.941) - winInputHook 
(3396):
Input: kb(desktop):leftArrow
DEBUG - editableText.EditableText._hasCaretMoved (08:35:39.982) - 
MainThread (1140):
Caret move detected using event. Elapsed 0 sec, retries 0
IO - speech.speech.speak (08:35:39.991) - MainThread (1140):
Speaking [CharacterModeCommand(True), '：', EndUtteranceCommand()]
IO - inputCore.InputManager.executeGesture (08:35:40.347) - winInputHook 
(3396):
Input: kb(desktop):leftArrow
DEBUG - editableText.EditableText._hasCaretMoved (08:35:40.386) - 
MainThread (1140):
Caret move detected using event. Elapsed 0 sec, retries 0
IO - speech.speech.speak (08:35:40.396) - MainThread (1140):
Speaking [CharacterModeCommand(True), '？', EndUtteranceCommand()]
IO - inputCore.InputManager.executeGesture (08:35:40.757) - winInputHook 
(3396):
Input: kb(desktop):leftArrow
DEBUG - editableText.EditableText._hasCaretMoved (08:35:40.806) - 
MainThread (1140):
Caret move detected using event. Elapsed 0 sec, retries 0
IO - speech.speech.speak (08:35:40.812) - MainThread (1140):
Speaking [CharacterModeCommand(True), '！', EndUtteranceCommand()]
IO - inputCore.InputManager.executeGesture (08:35:41.196) - winInputHook 
(3396):
Input: kb(desktop):leftArrow
DEBUG - editableText.EditableText._hasCaretMoved (08:35:41.237) - 
MainThread (1140):
Caret move detected using event. Elapsed 0 sec, retries 0
IO - speech.speech.speak (08:35:41.246) - MainThread (1140):
Speaking [CharacterModeCommand(True), '．', EndUtteranceCommand()]
IO - inputCore.InputManager.executeGesture (08:35:41.636) - winInputHook 
(3396):
Input: kb(desktop):leftArrow
DEBUG - editableText.EditableText._hasCaretMoved (08:35:41.672) - 
MainThread (1140):
Caret move detected using event. Elapsed 0 sec, retries 0
IO - speech.speech.speak (08:35:41.686) - MainThread (1140):
Speaking ['crochet gauche', EndUtteranceCommand()]

**Comment by CyrilleB79** [comment]
There are many characters that are not spoken by OneCore or Eloquence. In the first place, it would be the job of the synth to report them. On the contrary, eSpeak reports something.

NVDA takes into account that not all synth correctly report symbols and provide the symbols.dic file / symbols dialog to overcome this problem.

Of course, we can just add the missing symbols in the symbol file.

The use case of reading a regexp in the advanced settings panel is clearly not the most significant one. There are probably other open tickets asking to add more symbols in the symbol file.

The question is:
To what extent should the symbol file compensate for the problems of the synths?
Is there a risk if we add too much symbols in this file, e.g. performance issue?

@seanbudd, NV Access' opinion on this (if any) would be appreciated.

If this issue is implemented updating the symbol file, care should be taken to not break Chinese text reading at any level of punctuation. Probably preserve=all should be used.


The question is:

**Comment by seanbudd** [comment]
I don't think there is a risk to adding too many symbols to this file, I think we should set sensible defaults for as many symbols as deemed appropriate.
I think since one main purpose of the symbol file is to compensate for synthesizer behaviour, adding more symbols where appropriate makes sense.
